spec/factories/registration_payment.rb

```diff
@@ -20,16 +21,22 @@
       amount_lowest_denomination { competition.base_entry_fee_lowest_denomination * 2 }
     end
 
-    trait :skip_create_hook do
+    trait :skip_create_hook_bulk do
       skip_auto_accept_hook { true }
+      revert_mode_to { :bulk }
+    end
+
+    trait :skip_create_hook_live do
+      skip_auto_accept_hook { true }
+      revert_mode_to { :live }
     end
 
     after(:build) do |_payment, evaluator|
-      evaluator.competition.auto_accept_registrations = false if evaluator.skip_auto_accept_hook
+      evaluator.competition.auto_accept_preference = :disabled if evaluator.skip_auto_accept_hook
     end
 
     after(:create) do |_payment, evaluator|
-      evaluator.competition.auto_accept_registrations = true if evaluator.skip_auto_accept_hook
+      evaluator.competition.auto_accept_preference = evaluator.revert_mode_to if evaluator.skip_auto_accept_hook
```

Comment:
Just curious: Does `evaluator.competition.auto_accept_preference_previously_was` work?

Reply:
If yes, that would make the whole `revert_mode_to` thingy obsolete